### ISSUE TITLE
Winbtrfs not stable for gaming+modding. 

### ISSUE THREAD

**Issue by jarrard** [maharmstone/btrfs#527]
Just wanted to let people know that winbtrfs is not stable for gaming on or modding. 

I just finished spending days testing modding with Mod Organizer 2 _(likely other tools affected that use windows virtual filesystem toolkit etc..)_ and having issues ranging from mods just behaving corrupted or MO2 having 2minute pauses in-between installing mods.. 

Basically was happening whenever the software touched a btrfs drive. I also tried turning off compression, no difference. 

Games can work mostly fine by themselves but as soon as you start using windows .net and other tool api's, things can get real spicy!  

Just a warning for anyone having similar issues or thinking btrfs might be a good solution to move secondary drives to for running and storing games or applications.

**Comment by maharmstone** [comment]
Not a particularly helpful title...

If you want me to look into this, please provide:
* The version of Mod Organizer 2 you're using
* The exact version of Windows you're on (`winver`)
* Which games show the problem, and
* Which mods show the problem

And I'll try to reproduce.

**Comment by jarrard** [comment]
Here is some more info. Problem is not specific to any particular mod or game, but in general.

- MO2 2.3.4-2.4.4 (likely more)
- Windows 11 Insider AND Windows 10 (19044.2075)
- This is not a game problem but more a MO2 UI issue which passes on to games but I was testing with FO4
- ANY mods... I just used F4EE from WJ but you can just add in any mod to download list to trigger the freeze provided its stored on btrfs

**Comment by maharmstone** [comment]
Thanks, I'll have a look. MO2 is doing some voodoo in https://github.com/ModOrganizer2/usvfs, which might be related.

**Comment by jarrard** [comment]
yeah except the usvfs system is only meant to mount anything on launch. The issues occur at program start, unless usvfs is loaded into memory beforehand.. still weird.

**Comment by jarrard** [comment]
Ok well I did this on Win11 and Win10, new installs. I had compress turned off but its possible some of the files were still in zstd mode since it was enabled at some point on the drives.  Maybe that is the missing part?

These were NVMe and a normal HDD (for mod storage)

Also running MO2 off the BTRFS resulted in corrupt files somehow and I had to reinstall the modpack I was testing.  

Either way back on NTFS now and ALL ISSUES gone!

**Comment by psisis** [comment]
Hmm. I have previously referred to how Fallout: NV would not launch if its mods were copied unto the Btrfs volume. I was using Mod Organizer 2 as well, but I don’t have a crash-dump and maybe 1.8.1 has fixed this. I will recopy to Btrfs and repro.

**Comment by stefanverbeeck** [comment]
I am having similar issues using MO2.
Some operations in the application cause a BSOD with error "KERNEL_SECURITY_CHECK_FAILURE".
I cannot find what is exectly triggering it, but usually it involved sending something to the recycle bin from within the application. 

Running Win10 version 19045.2486, no secure boot.
MO2 version 2.4.4, using btrfs 1.8.1
The main folder of that tool is stored on the btrfs drive.

**Comment by maharmstone** [comment]
What about 1.8.2?

Also, please attach a minidump for your BSOD.

**Comment by stefanverbeeck** [comment]
I just tried 1.8.2 and that one also gave a BSOD.
This time, the message stated 'SYSTEM_SERVICE_EXCEPTION', what failed: btrfs.sys

The curious thing is that this crash happened when I was not even doing anything in the MO2 program.
[011123-8265-01.zip](https://github.com/maharmstone/btrfs/files/10394350/011123-8265-01.zip)

**Comment by maharmstone** [comment]
@stefanverbeeck, can you please paste the output of `btrfs check` on Linux?

**Comment by stefanverbeeck** [comment]
Hi, Sorry for the delay.
Below the output of btrrfs check:
Checking filesystem on /dev/sda6
UUID: 5409a8a3-04f3-40d2-87de-de2ca889a570
[1/7] checking root items
[2/7] checking extents
[3/7] checking free space cache
[4/7] checking fs roots
[5/7] checking only csums items (without verifying data)
[6/7] checking root refs
[7/7] checking quota groups skipped (not enabled on this FS)
found 422573105152 bytes used, no error found
total csum bytes: 411394136
total tree bytes: 1190952960
total fs tree bytes: 462716928
total extent tree bytes: 234995712
btree space waste bytes: 167998256
file data blocks allocated: 445267451904
 referenced 545024311296
